Widgets/Controls/Counter.ascx.cs

```diff
@@ -0,0 +1,80 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Web;
+using System.Web.UI;
+using System.Web.UI.WebControls;
+
+namespace Widgets.Controls {
+    public partial class Counter : System.Web.UI.UserControl {
+
+        public int counter {
+            get{
+                object o = ViewState["counter"];
+                if (o == null) {
+                    return 0;
+                }
+                else {
+                    return (int)o;
+                }
+            } 
+            set{
+                ViewState["counter"] = value;
+            }
+        }
+
+        void Decrement_Click(object sender, EventArgs e) {
+            counter -= 1;
+            number.Text = counter.ToString();
+        }
+        void Increment_Click(object sender, EventArgs e) {
+            counter += 1;
+            number.Text = counter.ToString();
+        }
+
+        protected void Page_Load(object sender, EventArgs e) {
+            if (IsPostBack) {
+                decrement.Click += new EventHandler(this.Decrement_Click);
+                increment.Click += new EventHandler(this.Increment_Click);
+            }
+            else {
+                number.Text = counter.ToString();
```

Comment:
I'd recommend adding a `Page_PreRender` event handler method and move this line of code into that method:

```
number.Text = counter.ToString();
```

Then you could remove this line of code from your `Page_Load` method and each of your button click event handler methods.